share/completions/fish_opt.fish

```diff
@@ -0,0 +1,12 @@
+set -g CONDITION '! __fish_seen_argument -s r --long required-val -s o --long optional-val'
```

Comment:
This makes a global variable called "CONDITION", visible outside of the script. That means if you had a variable called CONDITION it would change it.

Make it local instead - `set -l CONDITION`.